documentation/reference/tensor.html

```diff
@@ -118,23 +119,71 @@ <h3 id="literal-form-examples">Literal form examples</h3>
 <pre>
 tensor&lt;float&gt;(x[3]):{ {x:0}:3.0, {x:1}:5.0, {x:2}:7.0 }
 </pre>
-The same tensor in dense form:
+A tensor with a type using the default value type (double) and quoted labels:
+<pre>
+tensor(key{}):{ {key:'key.1'}:3.0, {key:'key 2'}:5.0, {key:"key's"}:7.0 }
+</pre>
+</p>
+
+
+<h3 id="indexed-short-form">Indexed short form</h3>
+
+<p>Tensors where all dimensions are indexed can be written as numbers wrapped in square brackets in
+<i>right dimension adjacent</i> order. This form requires an explicit tensor type.</p>
+
+<p>Brackets must either be nested according to the structure of the type, where values in dimensions to the
+right are closer than dimensions on the left, or alternatively all values may be given (in the same order)
+as a flat array (the latter not supported in expressions).</p>
+
+<b>Indexed short form examples:</b>
+
+<p>
+A float 1d tensor in indexed form:
 <pre>
 tensor&lt;float&gt;(x[3]):[3.0, 5.0, 7.0]
 </pre>
-A matrix in dense form. Note that the values for the right-most dimension (y) are adjacent, so e.g
+A matrix in indexed form. Since the values for the right-most dimension (y) are adjacent,
 the value 3 is here assigned to the cell {x:0,y:2}:
 <pre>
 tensor&lt;float&gt;(x[2],y[3]):[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]
 </pre>
-Also note that left and right dimension here is by alphabetical order of the dimension names, the order in which
-they are given in the type specification is irrelevant. The inner brackets are syntactic sugar and can
-be omitted, so the above is equivalent to
+The dimension order here is given by the order in which the dimensions are specified, not in the
+natural (alphabetic) order inherent in the tensor type.
+Since inner brackets can be omitted, the above is equivalent to
 <pre>
 tensor&lt;float&gt;(x[2],y[3]):[1.0, 2.0, 3.0, 4.0, 5.0, 6.0]
 </pre>
 </p>
 
+<h3 id="map-short-form">Map short form</h3>
+
+<p>Tensors with a single mapped dimension can be written by specifying just the label in that implicit dimension
+instead of a full address map. This form requires a type to be specified.</p>
+
+<b>Map short form example:</b>
+
+<p>
+<pre>
+tensor&lt;float&gt;(key{}):{ key1:1.0
+                       key2:2.0 }
```

Comment:
in the back-end, we require comma before key2

Reply:
Same